src/data.ts

```diff
@@ -55,42 +55,26 @@ export enum Effectiveness {
   ZERO = 0
 }
 
-function rawDataStrToNumber(str: string): Effectiveness {
-  if (str === "2") {
-    return Effectiveness.DOUBLE;
-  }
-  if (str === "1") {
-    return Effectiveness.REGULAR;
-  }
-  if (str === "½") {
-    return Effectiveness.HALF;
-  }
-  if (str === "0") {
-    return Effectiveness.ZERO;
-  }
-  throw new Error();
-}
-
 const rawData = [
-  "1 1 1 1 1 ½ 1 0 ½ 1 1 1 1 1 1 1 1 1",
-  "2 1 ½ ½ 1 2 ½ 0 2 1 1 1 1 ½ 2 1 2 ½",
-  "1 2 1 1 1 ½ 2 1 ½ 1 1 2 ½ 1 1 1 1 1",
-  "1 1 1 ½ ½ ½ 1 ½ 0 1 1 2 1 1 1 1 1 2",
-  "1 1 0 2 1 2 ½ 1 2 2 1 ½ 2 1 1 1 1 1",
-  "1 ½ 2 1 ½ 1 2 1 ½ 2 1 1 1 1 2 1 1 1",
-  "1 ½ ½ ½ 1 1 1 ½ ½ ½ 1 2 1 2 1 1 2 ½",
-  "0 1 1 1 1 1 1 2 1 1 1 1 1 2 1 1 ½ 1",
-  "1 1 1 1 1 2 1 1 ½ ½ ½ 1 ½ 1 2 1 1 2",
-  "1 1 1 1 1 ½ 2 1 2 ½ ½ 2 1 1 2 ½ 1 1",
-  "1 1 1 1 2 2 1 1 1 2 ½ ½ 1 1 1 ½ 1 1",
-  "1 1 ½ ½ 2 2 ½ 1 ½ ½ 2 ½ 1 1 1 ½ 1 1",
-  "1 1 2 1 0 1 1 1 1 1 2 ½ ½ 1 1 ½ 1 1",
-  "1 2 1 2 1 1 1 1 ½ 1 1 1 1 ½ 1 1 0 1",
-  "1 1 2 1 2 1 1 1 ½ ½ ½ 2 1 1 ½ 2 1 1",
-  "1 1 1 1 1 1 1 1 ½ 1 1 1 1 1 1 2 1 0",
-  "1 ½ 1 1 1 1 1 2 1 1 1 1 1 2 1 1 ½ ½",
-  "1 2 1 ½ 1 1 1 1 ½ ½ 1 1 1 1 1 2 2 1"
-].map(row => row.split(" ").map(rawDataStrToNumber));
+  [1, 1, 1, 1, 1, 0.5, 1, 0, 0.5, 1, 1, 1, 1, 1, 1, 1, 1, 1],
+  [2, 1, 0.5, 0.5, 1, 2, 0.5, 0, 2, 1, 1, 1, 1, 0.5, 2, 1, 2, 0.5],
+  [1, 2, 1, 1, 1, 0.5, 2, 1, 0.5, 1, 1, 2, 0.5, 1, 1, 1, 1, 1],
+  [1, 1, 1, 0.5, 0.5, 0.5, 1, 0.5, 0, 1, 1, 2, 1, 1, 1, 1, 1, 2],
+  [1, 1, 0, 2, 1, 2, 0.5, 1, 2, 2, 1, 0.5, 2, 1, 1, 1, 1, 1],
+  [1, 0.5, 2, 1, 0.5, 1, 2, 1, 0.5, 2, 1, 1, 1, 1, 2, 1, 1, 1],
+  [1, 0.5, 0.5, 0.5, 1, 1, 1, 0.5, 0.5, 0.5, 1, 2, 1, 2, 1, 1, 2, 0.5],
+  [0, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 2, 1, 1, 0.5, 1],
+  [1, 1, 1, 1, 1, 2, 1, 1, 0.5, 0.5, 0.5, 1, 0.5, 1, 2, 1, 1, 2],
+  [1, 1, 1, 1, 1, 0.5, 2, 1, 2, 0.5, 0.5, 2, 1, 1, 2, 0.5, 1, 1],
+  [1, 1, 1, 1, 2, 2, 1, 1, 1, 2, 0.5, 0.5, 1, 1, 1, 0.5, 1, 1],
+  [1, 1, 0.5, 0.5, 2, 2, 0.5, 1, 0.5, 0.5, 2, 0.5, 1, 1, 1, 0.5, 1, 1],
+  [1, 1, 2, 1, 0, 1, 1, 1, 1, 1, 2, 0.5, 0.5, 1, 1, 0.5, 1, 1],
+  [1, 2, 1, 2, 1, 1, 1, 1, 0.5, 1, 1, 1, 1, 0.5, 1, 1, 0, 1],
+  [1, 1, 2, 1, 2, 1, 1, 1, 0.5, 0.5, 0.5, 2, 1, 1, 0.5, 2, 1, 1],
+  [1, 1, 1, 1, 1, 1, 1, 1, 0.5, 1, 1, 1, 1, 1, 1, 2, 1, 0],
+  [1, 0.5, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 2, 1, 1, 0.5, 0.5],
+  [1, 2, 1, 0.5, 1, 1, 1, 1, 0.5, 0.5, 1, 1, 1, 1, 1, 2, 2, 1]
+];
```

Comment:
Hah, thanks. I forgot I did it like this. I had a script that copied that table from the DOM on Bulbapedia, and I never bothered to just manually convert it.